R/ospPlotConfiguration.R

```diff
@@ -0,0 +1,403 @@
+#' @title Convenience class for plot configuration for OSP plots
+#'
+#' @description
+#'
+#' R6 class defining the configuration for `{ospsuite}` plots. It holds values
+#' for all relevant plot properties.
+#'
+#' Note that the values this objects contains are of general-purpose utility. In
+#' other words, the public members of this class instance can be used to specify
+#' arguments for base plots, ggplot plots, or any other plotting framework.
+#'
+#' @field title,subtitle,xlabel,ylabel,legendTitle,watermark A character string
+#'   providing plot annotations for plot title, subtitle, x-axis label, y-axis
+#'   label, plot legend, watermark, respectively.
+#' @field legendPosition A character string defining the legend position
+#'   (default: `"outsideRight"`). Available options
+#'   are:`"none"`,`"insideTop"`,`"insideTopLeft"`,
+#'   `"insideLeft"`,`"insideBottomLeft"`,`"insideBottom"`,`"insideBottomRight"`,
+#'   `"insideRight"`,`"insideTopRight"`,`"outsideTop"`,`"outsideTopLeft"`,
+#'   `"outsideLeft"`,`"outsideBottomLeft"`,`"outsideBottom"`,`"outsideBottomRight"`,
+#'    `"outsideRight"`,`"outsideTopRight"`.
```

Comment:
add hint that all options are accessible via `tlf::LegendPositions` e.g. `tlf::LegendPositions$insideLeft`
https://www.open-systems-pharmacology.org/TLF-Library/reference/LegendPositions.html